libs/dmlf/tests/execution/basic_engine_tests.cpp

```diff
@@ -0,0 +1,860 @@
+
+  ExecutionResult createdState = engine.CreateState("state");
+  EXPECT_TRUE(createdState.succeeded());
+
+  ExecutionResult result = engine.Run("helloWorld", "state", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 1);
+}
+
+TEST(BasicVmEngineDmlfTests, DoubleHelloWorld)
+{
+  BasicVmEngine engine;
+
+  ExecutionResult createdProgram = engine.CreateExecutable("helloWorld", {{"etch", helloWorld}});
+  EXPECT_TRUE(createdProgram.succeeded());
+
+  ExecutionResult createdState = engine.CreateState("state");
+  EXPECT_TRUE(createdState.succeeded());
+
+  ExecutionResult result = engine.Run("helloWorld", "state", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 1);
+
+  result = engine.Run("helloWorld", "state", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 1);
+}
+
+TEST(BasicVmEngineDmlfTests, repeated_HelloWorld)
+{
+  BasicVmEngine engine;
+
+  ExecutionResult createdProgram = engine.CreateExecutable("helloWorld", {{"etch", helloWorld}});
+  EXPECT_TRUE(createdProgram.succeeded());
+
+  ExecutionResult createdState = engine.CreateState("state");
+  EXPECT_TRUE(createdState.succeeded());
+
+  createdProgram = engine.CreateExecutable("helloWorld", {{"etch", helloWorld}});
+  createdState   = engine.CreateState("state");
+
+  EXPECT_FALSE(createdProgram.succeeded());
+  EXPECT_EQ(createdProgram.error().stage(), Stage::ENGINE);
+  EXPECT_EQ(createdProgram.error().code(), Code::BAD_EXECUTABLE);
+  EXPECT_FALSE(createdState.succeeded());
+  EXPECT_EQ(createdState.error().stage(), Stage::ENGINE);
+  EXPECT_EQ(createdState.error().code(), Code::BAD_STATE);
+
+  ExecutionResult result = engine.Run("helloWorld", "state", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 1);
+}
+
+TEST(BasicVmEngineDmlfTests, Tick_2States)
+{
+  BasicVmEngine engine;
+
+  ExecutionResult createdProgram = engine.CreateExecutable("tick", {{"etch", tick}});
+  EXPECT_TRUE(createdProgram.succeeded());
+
+  ExecutionResult createdState = engine.CreateState("state1");
+  EXPECT_TRUE(createdState.succeeded());
+  createdState = engine.CreateState("state2");
+  EXPECT_TRUE(createdState.succeeded());
+
+  ExecutionResult result = engine.Run("tick", "state1", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 0);
+
+  result = engine.Run("tick", "state1", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 1);
+
+  result = engine.Run("tick", "state2", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 0);
+
+  result = engine.Run("tick", "state1", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 2);
+
+  result = engine.Run("tick", "state2", "main");
+  EXPECT_TRUE(result.succeeded());
+  EXPECT_EQ(result.output().Get<int>(), 1);
+}
+
+// Repeated things
+
+// Breaks VM
```

Comment:
😢

Reply:
Yeah Robert knows and told he would look into a fix